src/H5C.c

```diff
@@ -6579,8 +6579,10 @@ H5C__make_space_in_cache(H5F_t *f, size_t space_needed, hbool_t write_permitted)
     H5C_cache_entry_t *entry_ptr;
     H5C_cache_entry_t *prev_ptr;
     H5C_cache_entry_t *next_ptr;
-    uint32_t           num_corked_entries = 0;
-    herr_t             ret_value          = SUCCEED; /* Return value */
+#ifndef NDEBUG
+    uint32_t num_corked_entries = 0;
+#endif                          /* NDEBUG */
```

Comment:
We should probably try to use H5C_DEBUG instead of NDEBUG for package-level debugging, as we do elsewhere in the library